Phalcon 3.4.x on PHP 8

Bellardia opened the pull request:
Introduces the required Packer changes to build PHP 8 (and 7.4) and MySql 5.7 on Ubuntu Bionic and Ubuntu Focal.

We are limited to running Ubuntu Bionic in AWS OpsWorks, so that will continue to be our base environment for the time being, however I ported everything over to Ubuntu Focal (20.04) as well and it's possible to use such images everywhere except in AWS OpsWorks at the moment (in theory the work in raw EC2, it's namely the `OpsWorks Agent` does not support anything `> 18.04` so instances cannot be managed by OpsWorks).

Bellardia committed:
Implement PHP8

Bellardia committed:
Fix MySql 5.7 for 18.04/20.04

Bellardia committed:
Support for PHP8 / FREEXIAN

Bellardia committed:
Use suitable Phalcon branch for PHP version

Bellardia committed:
Change name of PHP socket to depend on php_version

Bellardia committed:
Support for Jammy 22.04

ahosking commented on base/recipes/mysql.rb:
```diff
@@ -28,11 +35,32 @@
+    notifies :disable, "service[mysql]", :immediately if pkg == "mysql-server"
+  end
+end
+
```
@Bellardia is this the only reliable way to pin our versions and keep mysql humming? 
Storing the binaries seems like a bummer, but, necessary.
I recall from the last time I worked on this that getting the repos from other distribution versions did cause more problems than it solved.

Bellardia replied:
yes, the dependencies can't be built on 18.04 due to other missing dependencies. I need to use pre-built versions from other sources

there's a way to do it by installing through mysql 8 but it puts the system in a worse state overall with different dep issues

ahosking commented on configure/templates/default/my.cnf.erb:
```diff
@@ -170,9 +169,9 @@ sort_buffer_size                = 1M    # UPD
-table_definition_cache          = 50000 # UPD
-table_open_cache                = 50000 # UPD
-open_files_limit                = 100000 # UPD - This can be 2x to 3x the table_open_cache value or match the system's
+table_definition_cache          = 32713 # UPD
+table_open_cache                = 32713 # UPD
+open_files_limit                = 65565 # UPD - This can be 2x to 3x the table_open_cache value or match the system's
```
What is the significance of these new, lower values?

Bellardia replied:
pretty sure they're just under the default open files limit on docker which is less than they were on vagrant. 
as to what level of the stack is enforcing them im not sure, but the numbers are high enough that it's not especially consequential

Bellardia committed:
Add ubuntu archive apt repo for libboost-1.71.0

Bellardia committed:
Upgrade to node18 LTS

Bellardia committed:
Only side-load libssl dep for jammy+

Bellardia committed:
Add Datadog support to Docker

Bellardia committed:
Tune settings for PHP 8 (#86)

Bellardia committed:
feat: disable auto-ssl in dev

Auto SSL adds a ton of extra latency if the node is not reachable by external traffic

Bellardia committed:
feat: fix supervisorctl configuration for dnsmasq & datadog

Bellardia committed:
fix: ensure gearman database table name is always valid

Bellardia committed:
feat: allow private networking in dev

Bellardia committed:
feat: build the most recent version of dnsmasq

Bellardia committed:
chore: bump hashi product versions to latest releases

Bellardia committed:
feat: improve key derivation for limit_conn_zone to correctly bucket sessions

Bellardia committed:
chore: set default php version to 8.2
